### PR TITLE
Datascience feature flag

### PR DESCRIPTION
**Pull request by IanMatthewHuff**
Place data science behind a feature flag.
Only handle shift+enter if we have code cells

### CODE REVIEW

File: src/client/datascience/datascience.ts [hunk 1]
```diff
@@ -18,20 +19,26 @@ export class DataScience implements IDataScience {
...
-        // We could potentially register different commands if jupyter isn't installed.
+        this.configuration = this.serviceContainer.get<IConfigurationService>(IConfigurationService);
     }
 
     public async activate(): Promise<void> {
         this.registerCommands();
 
+        // Check if data science is disabled, if so, turn off our editor context to hide commands
+        const settings = this.configuration.getSettings();
+        const enabled = settings.datascience.enabled;
```
Comment (rchiodo) [inline src/client/datascience/datascience.ts]: >enabled [](start = 14, length = 7)

What happens if they suddenly enable the setting during the vs code running? Do they need to restart?

Comment (IanMatthewHuff) [reply src/client/datascience/datascience.ts]: Yeah, that's how I was treating it. Might be worth seeing if there is an easy way to pick up the setting change. I'll take a quick peek.

File: src/client/datascience/constants.ts [hunk 2]
```diff
@@ -14,6 +14,7 @@ export namespace Commands {
 
 export namespace EditorContexts {
     export const HasCodeCells = 'python.datascience.hascodecells';
+    export const DataScienceEnabled = 'python.datascience.enabledcontext';
```
Comment (DonJayamanne) [inline src/client/datascience/constants.ts]: Why the suffix `context`, we haven't used it in other context Jerry's, I'd remove it to be consistent

Comment (IanMatthewHuff) [reply src/client/datascience/constants.ts]: @DonJayamanne I had the setting named 'python.datascience.enabled' and I didn't want to conflict the context and the setting name for searching.

Comment (IanMatthewHuff) [reply src/client/datascience/constants.ts]: I guess I could do .featureenabled or something like that to avoid the conflict.

**Pull request merged**